### ISSUE TITLE
Default HCal veto settings ignores a large portion of the back HCal

### ISSUE THREAD

**Issue by EinarElen** [LDMX-Software/Hcal#61]
The default settings for the `HcalVetoProcessor` will ignore any hit where `z > 4 m`. The v14 Hcal is ~4.7 meters in depth (see [gdml](https://github.com/LDMX-Software/ldmx-sw/blob/8eca731541a8a1e5df23a01482a6d04f665cd059/Detectors/data/ldmx-det-v14/constants.gdml#L174-L179)) so parts the back HCal currently isn't used.
```python 
    def __init__(self,name = 'hcalVeto') :
        super().__init__(name,'hcal::HcalVetoProcessor','Hcal')
        self.pe_threshold = 5.0
        self.max_time = 50.0
        self.max_depth = 4000.0
        self.back_min_pe = 1.
```
```C++
    if (hcalHit.getZPos() > maxDepth_) continue;
```

I'm guessing that this isn't what we want. This could also explain why we sometimes trigger the bug in #58. I'm guessing that this configuration parameter is a remain of when the back depth was configurable. If we still want this to be an option, I would say that
- The location start of the HCal should be taken into account in the veto processor (not sure where it is now)
- The default max depth should be given by the geometry condition rather than as a parameter 
We could still have an option to set a different max_depth but leave the default to the geometry condition.

**Comment by bryngemark** [comment]
I think this setting is a remainder from when in `v9`, the cal was 6m deep in the gdml. This was good for inefficiency studies but wasn't considered realistic given space constraints in ESA, so the veto didn't consider the last two meters. I agree that having a default max depth setting in meters is confusing and error prone. 

One might have a default which maps to using the full depth as given by the geometry (`-1` or something) and use that to define a max depth in initialization, while still allowing setting some actual physical max depth if that is desirable (e.g. for veto efficiency studies with a thinner Hcal).

**Comment by EinarElen** [comment]
Closed by #70
